lib/asset_sources.rb

```diff
@@ -1,54 +1,67 @@
 # frozen_string_literal: true
 
 class AssetSources
-  class << self
-    attr_accessor :manifest_path
-    attr_accessor :manifest
-    attr_accessor :cache_manifest
-
-    def get_sources(*names)
-      # RailsI18nWebpackPlugin will generate additional assets suffixed per locale, e.g. `.fr.js`.
-      # See: app/javascript/packages/rails-i18n-webpack-plugin/extract-keys-webpack-plugin.js
-      regexp_locale_suffix = %r{\.(#{I18n.available_locales.join('|')})\.js$}
-
-      load_manifest_if_needed
-
-      locale_sources, sources = names.flat_map do |name|
-        manifest&.dig('entrypoints', name, 'assets', 'js')
-      end.uniq.compact.partition { |source| regexp_locale_suffix.match?(source) }
-
-      [
-        *locale_sources.filter { |source| source.end_with? ".#{I18n.locale}.js" },
-        *sources,
-      ]
-    end
+  attr_reader :manifest_path
+  attr_reader :manifest
+  attr_reader :cache_manifest
 
-    def get_assets(*names)
-      load_manifest_if_needed
+  def initialize(manifest_path:, cache_manifest:, i18n_locales:)
+    @manifest_path = manifest_path
+    @cache_manifest = cache_manifest
+    @regexp_locale_suffix = %r{\.(#{i18n_locales.join('|')})\.js$}
 
-      names.flat_map do |name|
-        manifest&.dig('entrypoints', name, 'assets')&.except('js')&.values&.flatten
-      end.uniq.compact
+    if cache_manifest
```

Comment:
This is a slight chance in behavior to avoid changing the values of the instance outside of instantiation